task-driver/integration/helpers.rs

```diff
@@ -1,9 +1,62 @@
 //! Helpers for `task-driver` integration tests
 
-use common::types::wallet_mocks::mock_empty_wallet;
+use circuit_types::{
+    native_helpers::create_wallet_shares_from_private, traits::BaseType, SizedWalletShare,
+};
+use common::types::{
+    proof_bundles::mocks::{dummy_valid_wallet_create_bundle, dummy_valid_wallet_update_bundle},
+    wallet::Wallet,
+    wallet_mocks::mock_empty_wallet,
+};
 use external_api::types::Wallet as ApiWallet;
+use eyre::Result;
+use mpc_stark::algebra::scalar::Scalar;
+use rand::thread_rng;
+use renegade_crypto::hash::{evaluate_hash_chain, PoseidonCSPRNG};
+use starknet_client::client::StarknetClient;
 use system_bus::SystemBus;
-use task_driver::driver::TaskDriver;
+use task_driver::driver::{TaskDriver, TaskDriverConfig};
+
+// ------------------------
+// | Contract Interaction |
+// ------------------------
+
+/// Create a wallet in the contract state
+pub async fn allocate_wallet_in_darkpool(wallet: &Wallet, client: &StarknetClient) -> Result<()> {
```

Comment:
thoughts on consolidating this with `deploy_new_wallet` in `starknet-client/integration/helpers.rs`? don't have a strong opinion here, just seems like these two helpers do allllmost the same thing

Reply:
Their implied interface is different enough for them to be separate; I think combining them would be over-optimizing code-reuse at the expense of complicating the interaction with the helper in both cases.